### ISSUE TITLE
Error deploying .NET WebService which uses CamlBuilder v1.1.1.0 

### ISSUE THREAD

**Issue by gsalatino** [joaope/camlbuilder#15]
Hi, 

We've developed a Remote Event Receiver and part of it uses CamlBuilder 1.1.0.0. After update its references to CamlBuilder 1.1.1.0 we couldn't redeploy it in Azure Portal. We get the following error:

Exception type: ConfigurationErrorsException
Exception message: Could not load file or assembly 'CamlBuilder, Version=1.1.1.0, Culture=neutral, PublicKeyToken=f8277973dee80c28' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)

It seems that some step is missing in the process of signing the assembly.

This piece of software has been very helpful!
We would appreciate very much if you can review this problem.

kindest regards,
--
GS

**Comment by joaope** [comment]
Just deployed a new version (1.1.2) to NuGet. Please try it to see if it helps. Otherwise I'll need to look at it properly as soon as I've the time.

**Comment by gsalatino** [comment]
Hi Joa! 
Thanks a lot for your prompt response!
We'll try it right now and see if the setting <DelaySign>false</DelaySign> solve the problem.
I'll keep you posted.
Best regards,
--
GS

**Comment by gsalatino** [comment]
Hi Joa, 

Unfortunately the new version 1.1.2.0 didn't resolve the FileLoadException. We still get the error when try to execute a web method of a .NET WCF web service. Here is the detailed info of the exception we captured:

Server Error in '/' Application.
Could not load file or assembly 'CamlBuilder' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code. 

Exception Details: System.IO.FileLoadException: Could not load file or assembly 'CamlBuilder' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)

Source Error: 

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

Assembly Load Trace: The following information can be helpful to determine why the assembly 'CamlBuilder' could not be loaded.


=== Pre-bind state information ===
LOG: DisplayName = CamlBuilder
 (Partial)
WRN: Partial binding information was supplied for an assembly:
WRN: Assembly Name: CamlBuilder | Domain ID: 3
WRN: A partial bind occurs when only part of the assembly display name is provided.
WRN: This might result in the binder loading an incorrect assembly.
WRN: It is recommended to provide a fully specified textual identity for the assembly,
WRN: that consists of the simple name, version, culture, and public key token.
WRN: See whitepaper http://go.microsoft.com/fwlink/?LinkId=109270 for more information and common solutions to this issue.
LOG: Appbase = file:///C:/DevProjects/YPF Migración SGD-SGN/Git Hx Source/WebServices/ControlledCopy.WebService/
LOG: Initial PrivatePath = C:\DevProjects\YPF Migración SGD-SGN\Git Hx Source\WebServices\ControlledCopy.WebService\bin
Calling assembly : (Unknown).
===
LOG: This bind starts in default load context.
LOG: Using application configuration file: C:\DevProjects\YPF Migración SGD-SGN\Git Hx Source\WebServices\ControlledCopy.WebService\web.config
LOG: Using host configuration file: C:\Users\gsalatino\Documents\IISExpress\config\aspnet.config
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework\v4.0.30319\config\machine.config.
LOG: Policy not being applied to reference at this time (private, custom, partial, or location-based assembly bind).
LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/Temporary ASP.NET Files/vs/234bb346/70bcfc52/CamlBuilder.DLL.
LOG: Attempting download of new URL file:///C:/Windows/Microsoft.NET/Framework/v4.0.30319/Temporary ASP.NET Files/vs/234bb346/70bcfc52/CamlBuilder/CamlBuilder.DLL.
LOG: Attempting download of new URL file:///C:/DevProjects/YPF Migración SGD-SGN/Git Hx Source/WebServices/ControlledCopy.WebService/bin/CamlBuilder.DLL.
LOG: Using application configuration file: C:\DevProjects\YPF Migración SGD-SGN\Git Hx Source\WebServices\ControlledCopy.WebService\web.config
LOG: Using host configuration file: C:\Users\gsalatino\Documents\IISExpress\config\aspnet.config
LOG: Using machine configuration file from C:\Windows\Microsoft.NET\Framework\v4.0.30319\config\machine.config.
LOG: Post-policy reference: CamlBuilder, Version=1.1.2.0, Culture=neutral, PublicKeyToken=f8277973dee80c28
ERR: Failed to complete setup of assembly (hr = 0x80131045). Probing terminated.

Stack Trace: 


[FileLoadException: Could not load file or assembly 'CamlBuilder' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)]

[FileLoadException: Could not load file or assembly 'CamlBuilder, Version=1.1.2.0, Culture=neutral, PublicKeyToken=f8277973dee80c28' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)]
   System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +0
   System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +36
   System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, Evidence assemblySecurity, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks) +152
   System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean forIntrospection) +77
   System.Reflection.RuntimeAssembly.InternalLoad(String assemblyString, Evidence assemblySecurity, StackCrawlMark& stackMark, Boolean forIntrospection) +21
   System.Reflection.Assembly.Load(String assemblyString) +28
   System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +38

[ConfigurationErrorsException: Could not load file or assembly 'CamlBuilder, Version=1.1.2.0, Culture=neutral, PublicKeyToken=f8277973dee80c28' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)]
   System.Web.Configuration.CompilationSection.LoadAssemblyHelper(String assemblyName, Boolean starDirective) +738
   System.Web.Configuration.CompilationSection.LoadAllAssembliesFromAppDomainBinDirectory() +217
   System.Web.Configuration.CompilationSection.LoadAssembly(AssemblyInfo ai) +130
   System.Web.Compilation.BuildManager.GetReferencedAssemblies(CompilationSection compConfig) +170
   System.Web.Compilation.BuildManager.GetPreStartInitMethodsFromReferencedAssemblies() +92
   System.Web.Compilation.BuildManager.CallPreStartInitMethods(String preStartInitListPath, Boolean& isRefAssemblyLoaded) +290
   System.Web.Compilation.BuildManager.ExecutePreAppStart() +157
   System.Web.Hosting.HostingEnvironment.Initialize(ApplicationManager appManager, IApplicationHost appHost, IConfigMapPathFactory configMapPathFactory, HostingEnvironmentParameters hostingParameters, PolicyLevel policyLevel, Exception appDomainCreationException) +549

[HttpException (0x80004005): Could not load file or assembly 'CamlBuilder, Version=1.1.2.0, Culture=neutral, PublicKeyToken=f8277973dee80c28' or one of its dependencies. Strong name signature could not be verified.  The assembly may have been tampered with, or it was delay signed but not fully signed with the correct private key. (Exception from HRESULT: 0x80131045)]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +10044576
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +95
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +254

Version Information: Microsoft .NET Framework Version:4.0.30319; ASP.NET Version:4.7.2046.0

**Comment by joaope** [comment]
Should be fixed now with the new published package.

> Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
> Copyright (c) Microsoft Corporation.  All rights reserved.
> 
> Assembly 'CamlBuilder.dll' is valid

Feel free to re-open the issue if something's wrong.

**Comment by gsalatino** [comment]
Hi João, 

Thank you very much for your response.

One month ago we applied the following workaround: we downloaded CamlBuilder v1.1.2.0, removed everything related to assembly strong signing and compiled it locally to include it in our solution. The workaround works fine and then we were able to continue with our work. 👍 

As soon as we have the opportunity we'll include in the sprint the upgrade to last stable version of CamlBuilder and test it. We'll keep you posted about the result of our test.

Once again, thank you very much for your time. 💯

Kindests regards,
--GS
